### PR TITLE
SDK extension to support Ethereum

### PR DESCRIPTION
**Pull request by rranjan3**
1) New implementations for work_order & worker_registry
2) Add contract specific configs in tcf_connector.toml

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

### CODE REVIEW

File: sdk/avalon_sdk/ethereum/proxy/ethereum_work_order.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                    "failed", "Invalid request string {}"
+                    .format(work_order_request))
+
+            txn_hash = self.__contract_instance.functions.workOrderSubmit(
+                work_order_id, worker_id, requester_id, work_order_request
+                ).buildTransaction(
+                    self.__eth_client.get_transaction_params()
+                )
+            tx = self.__eth_client.execute_transaction(txn_hash)
+            return tx
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/proxy/ethereum_work_order.py]: tx -> txn_status

File: sdk/avalon_sdk/ethereum/proxy/ethereum_work_order.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                    binascii.hexlify(work_order_id).decode("utf8")):
+                logging.error("Invalid work order id {}".format(work_order_id))
+                return construct_message(
+                    "failed", "Invalid work order id {}".format(work_order_id))
+            txn_hash = self.__contract_instance.functions.workOrderComplete(
+                work_order_id, work_order_response).buildTransaction(
+                    self.__eth_client.get_transaction_params()
+                )
+            tx = self.__eth_client.execute_transaction(txn_hash)
+            return tx
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/proxy/ethereum_work_order.py]: tx -> txn_status

File: sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+                return construct_message(
+                    "failed", "Invalid organization id {}".format(org_id))
+            if not is_valid_hex_str(
+                    binascii.hexlify(application_id).decode("utf8")):
+                logging.error(
+                    "Invalid application id {}".format(application_id))
+                return construct_message(
+                    "failed",
+                    "Invalid application id {}".format(application_id))
+            lookupResult = self.__contract_instance.functions.workerLookUp(
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py]: lookupResult  -> lookup_result

File: sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        Retrieve the worker identified by worker id
+        """
+        if (self.__contract_instance is not None):
+            if not is_valid_hex_str(binascii.hexlify(
+                    worker_id).decode("utf8")):
+                logging.error("Invalid worker id {}".format(worker_id))
+                return construct_message(
+                    "failed", "Invalid worker id {}".format(worker_id))
+
+            workerDetails = self.__contract_instance.functions.workerRetrieve(
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py]: workerDetails -> worker_details

File: sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+            if not is_valid_hex_str(binascii.hexlify(org_id).decode("utf")):
+                logging.error("Invalid organization id {}".format(org_id))
+                return construct_message(
+                    "failed", "Invalid organization id {}".format(org_id))
+            if not is_valid_hex_str(
+                    binascii.hexlify(application_id).decode("utf8")):
+                logging.error("Invalid application id {}".format(org_id))
+                return construct_message(
+                    "failed", "Invalid application id {}".format(org_id))
+            lookupResult = self.__contract_instance.functions.workerLookUpNext(
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py]: lookupResult -> lookup_result

File: sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+        if (self.__contract_instance is not None):
+            if (is_valid_hex_str(binascii.hexlify(worker_id).decode("utf8"))
+                    is False):
+                logging.error("Invalid worker id {}".format(worker_id))
+                return None
+            if not _is_valid_json(details):
+                logging.error("Invalid details {}".format(details))
+                return None
+
+            txn_status_hash = self.__contract_instance.functions.workerUpdate(
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py]: Any specific reason why this variable is names as txn_status_hash where as other functions name this variable as txn_hash

Comment (rranjan3) [reply sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py]: No specific reason Manoj. Updated all the txn variables to be more appropriate.

File: sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py [hunk 7]
```diff
@@ -0,0 +1,243 @@
+# Copyright 2020 Intel Corporation
```
Comment (Ram-srini) [inline sdk/avalon_sdk/ethereum/proxy/ethereum_worker_registry.py]: Is this file newly created? We already have ethereum_worker_registry.py in ethereum, you can use that righ?

File: sdk/avalon_sdk/ethereum/proxy/ethereum_work_order.py [hunk 8]
```diff
@@ -0,0 +1,177 @@
+# Copyright 2020 Intel Corporation
```
Comment (Ram-srini) [inline sdk/avalon_sdk/ethereum/proxy/ethereum_work_order.py]: Can you please add unit test for this class.

File: sdk/avalon_sdk/ethereum/ethereum_worker_registry_list.py [hunk 9]
```diff
@@ -245,15 +245,15 @@ def registry_update(self, org_id, uri, sc_addr, app_type_ids):
...
                     org_id, uri, sc_addr,
                     app_type_ids).buildTransaction(
                         self.__eth_client.get_transaction_params()
                         )
-            txn_status = self.__eth_client.execute_transaction(
-                txn_statusn_hash)
-            return txn_status
+            txn_receipt = self.__eth_client.execute_transaction(
+                txn_dict)
+            return txn_receipt
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/ethereum_worker_registry_list.py]: function comments mentions Returns None on error. Does txn_receipt corresponds to error code ?

File: sdk/avalon_sdk/ethereum/proxy/ethereum_work_order.py (hunk 8 above)
Comment (rranjan3) [reply sdk/avalon_sdk/ethereum/proxy/ethereum_work_order.py]: I will be adding that and enabling all of them(including the 2 existing unit test files) in another PR.

**Pull request merged**